backend/routes/documents.py

```
@@ -308,9 +409,314 @@ async def upload_document(
+        logger.exception(
+            "Failed to invalidate study guides cache for user=%s course=%s",
+            user_id, course_id,
+        )
+
+
+def _check_upload_achievements(user_id: str) -> None:
+    """Background task: best-effort achievement check."""
+    try:
+        check_achievements(user_id, "documents_uploaded", {})
+    except Exception:
+        pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)